syntax = "proto3";
package bcos.protocol;

// Fields to calculate hash
message PBRawTransactionHashFields{
    int32 version = 1;
    bytes chainId = 2;
    bytes groupId = 3;
    int64 blockLimit = 4;
    bytes nonce = 5;
    bytes to = 6;
    bytes input = 7;
    bytes abi = 8;
}

message PBRawTransaction{
    bytes hashFieldsData = 1;
    bytes hashFieldsHash = 3;
    bytes signatureData = 2;
    int64 import_time = 12;
    uint32 attribute = 13;
    string source = 14;
}
